pkg/client/gen.go

```diff
@@ -375,5 +377,30 @@ func e2eManifest(cfg *GenConfig) *manifest.Manifest {
 		)
 	}
 
+	if e2eImageSupportsProgress(cfg.KubeConformanceImage) {
+		m.Spec.Env = updateExtraArgs(m.Spec.Env)
+	}
+
 	return m
 }
+
+func e2eImageSupportsProgress(imageName string) bool {
+	parts := strings.SplitAfter(imageName, ":")
+	tag := parts[len(parts)-1]
+	imageVersion, err := version.NewVersion(tag)
+	if err != nil {
+		return false
+	}
+	return imageVersion.GreaterThanOrEqual(version.Must(version.NewVersion(firstE2EVersionWithProgress)))
+}
+
+func updateExtraArgs(envs []corev1.EnvVar) []corev1.EnvVar {
+	for _, env := range envs {
+		// If set by user, just leave as-is.
+		if env.Name == "E2E_EXTRA_ARGS" {
+			return envs
+		}
+	}
+	envs = append(envs, corev1.EnvVar{Name: "E2E_EXTRA_ARGS", Value: "--progress-report-url=http://localhost:8099/progress"})
```

Comment:
Will have to double check where to grab the port from; I believe it is configurable.